apps/condo/domains/ticket/utils/serverSchema/propertyScope.js

```diff
@@ -0,0 +1,124 @@
+const { get } = require('lodash')
+const uniq = require('lodash/uniq')
+
+const { find } = require('@open-condo/keystone/schema')
+
+const { ORGANIZATION_TICKET_VISIBILITY, PROPERTY_TICKET_VISIBILITY, PROPERTY_AND_SPECIALIZATION_VISIBILITY } = require('@condo/domains/organization/constants/common')
+
+
+const _getPropertyAndSpecializationsEmployeesAccessedToTicket = async (propertyAndSpecDependsEmployeeInScope, ticketCategoryClassifier) => {
+    const availableToReadTicketUsers = []
+    const employeesWithAllSpecs = propertyAndSpecDependsEmployeeInScope.filter(employee => employee.hasAllSpecializations)
+
+    if (!ticketCategoryClassifier) {
+        availableToReadTicketUsers.push(
+            ...employeesWithAllSpecs.map(employee => employee.user),
+        )
+
+        return availableToReadTicketUsers
+    }
+
+    const employeeWithoutAllSpecs = propertyAndSpecDependsEmployeeInScope.filter(employee => !employee.hasAllSpecializations)
+    const employeeWithMatchedSpecialization = (await find('OrganizationEmployeeSpecialization', {
+        deletedAt: null,
+        employee: { id_in: employeeWithoutAllSpecs.map(employee => employee.id) },
+        specialization: { id: ticketCategoryClassifier },
+    })).map(obj => employeeWithoutAllSpecs.find(employee => employee.id === obj.employee))
+
+    availableToReadTicketUsers.push(
+        ...employeesWithAllSpecs.map(employee => employee.user),
+        ...employeeWithMatchedSpecialization.map(employee => employee.user)
+    )
+
+    return availableToReadTicketUsers
+}
+
+/**
+ * Finds user ids who available to read ticket by role and property scopes logic
+ * @param ticketOrganizationId
+ * @param ticketPropertyId
+ * @param ticketCategoryClassifierId
+ * @param ticketExecutorId
+ * @param ticketAssigneeId
+ * @returns {Promise<String>} User ids
+ */
+const getUsersAvailableToReadTicketByPropertyScope = async ({
+    ticketOrganizationId,
+    ticketPropertyId,
+    ticketCategoryClassifierId,
+    ticketExecutorId,
+    ticketAssigneeId,
+}) => {
+    const roles = await find('OrganizationEmployeeRole', {
+        organization: { id: ticketOrganizationId },
+        canReadTickets: true,
+    })
+    const organizationEmployees = await find('OrganizationEmployee', {
+        organization: { id: ticketOrganizationId },
+        deletedAt: null,
+        isRejected: false,
+        isAccepted: true,
+        isBlocked: false,
+    })
+    const employees = organizationEmployees.filter(
+        employee => roles.map(role => role.id).includes(employee.role)
```

Comment:
Each time filter call the callback, it perform `roles.map(role => role.id)`, although it only may be perform once)